packages/runtime/test/src/interpreter/interpreter.ts

```diff
@@ -0,0 +1,134 @@
+import { bobAcc } from "@algo-builder/algob/test/mocks/account";
+import { decodeAddress, getApplicationAddress } from "algosdk";
+import { assert } from "chai";
+
+import { AccountStore, getProgram, Interpreter, Runtime } from "../../../src";
+import RUNTIME_ERRORS from "../../../src/errors/errors-list";
+import { ALGORAND_ACCOUNT_MIN_BALANCE } from "../../../src/lib/constants";
+import { Stack } from "../../../src/lib/stack";
+import { AccountAddress, AccountStoreI, ExecutionMode, StackElem } from "../../../src/types";
+import { useFixture } from "../../helpers/integration";
+import { expectRuntimeError } from "../../helpers/runtime-errors";
+import { elonMuskAccount, johnAccount } from "../../mocks/account";
+import { accInfo } from "../../mocks/stateful";
+import { elonAddr, TXN_OBJ } from "../../mocks/txn";
+
+export function setDummyAccInfo(acc: AccountStoreI): void {
+	acc.appsLocalState = accInfo[0].appsLocalState;
+	acc.appsTotalSchema = accInfo[0].appsTotalSchema;
+	acc.createdApps = accInfo[0].createdApps;
+}
+
+describe("Interpreter", function () {
+	let interpreter: Interpreter;
+	let elonAcc: AccountStoreI;
+	let johnAcc: AccountStoreI;
+	let bobAccount: AccountStoreI;
+	let appAccAddr: AccountAddress;
+	let applicationAccount: AccountStoreI;
+	const elonPk = decodeAddress(elonAddr).publicKey;
+
+	const resetInterpreterState = (): void => {
+		while (interpreter.stack.length() !== 0) {
+			interpreter.stack.pop();
+		}
+		interpreter.currentInnerTxnGroup = [];
+		interpreter.runtime.ctx.pooledApplCost = 0;
+		interpreter.instructions = [];
+		interpreter.innerTxnGroups = [];
+		interpreter.instructionIndex = 0;
+		interpreter.runtime.ctx.tx = { ...TXN_OBJ, snd: Buffer.from(elonPk) };
+		interpreter.runtime.ctx.gtxs = [interpreter.runtime.ctx.tx];
+		interpreter.runtime.ctx.isInnerTx = false;
+		// set new tx receipt
+		interpreter.runtime.ctx.state.txReceipts.set(TXN_OBJ.txID, {
+			txn: TXN_OBJ,
+			txID: TXN_OBJ.txID,
+		});
+		interpreter.tealVersion = 1;
+	};
+
+	const setUpInterpreter = (
+		tealVersion: number,
+		appBalance: number = ALGORAND_ACCOUNT_MIN_BALANCE
+	): void => {
+		// setup 1st account (to be used as sender)
+		elonAcc = new AccountStore(0, elonMuskAccount); // setup test account
+		setDummyAccInfo(elonAcc);
+
+		// setup 2nd account
+		johnAcc = new AccountStore(0, johnAccount);
+
+		// setup 2nd account
```

Comment:
something wrong here

Reply:
Is it? I am just set up the account like the set up in the inner-transactions (https://github.com/scale-it/algo-builder/blob/11b6f9c318e80ad1b4ea866837a6b9777c4897fc/packages/runtime/test/src/interpreter/inner-transaction.ts#L52)